### PR TITLE
chore(ci): use nodejs 20 for datadog-ci

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by brettlangdon — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

File: .github/workflows/build-and-publish-image.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: Build and publish image
 
 on:
+  workflow_dispatch: {}
```
Comment (brettlangdon) [inline .github/workflows/build-and-publish-image.yml:4]:
```suggestion
```

This isn't useful here, we never run this workflow directly, it is included in other workflows, e.g. `testrunner.yml`.

File: docker/Dockerfile [hunk 2]
```diff
@@ -53,8 +53,10 @@ RUN apt-get update \
       awscli
 
 # Allow running datadog-ci in CI with npx
-RUN apt-get install -y --no-install-recommends nodejs npm \
-  && npm install -g @datadog/datadog-ci
+RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends nodejs npm \
```
Comment (brettlangdon) [inline docker/Dockerfile:58]:
```suggestion
    && apt-get install -y --no-install-recommends nodejs \
```

File: .github/workflows/build-and-publish-image.yml (hunk 1 above)
Comment (romainkomorndatadog) [reply .github/workflows/build-and-publish-image.yml]: Aaah! That's why I couldn't find my own runs of it. Derp. Thank you. :D

File: docker/Dockerfile (hunk 2 above)
Comment (romainkomorndatadog) [reply docker/Dockerfile]: why? AFAICT, npm isn't bundled with node?

File: docker/Dockerfile [hunk 3]
```diff
@@ -53,8 +53,10 @@ RUN apt-get update \
...
+    && npm install -g @datadog/datadog-ci
```
Comment (romainkomorndatadog) [inline docker/Dockerfile]:
```suggestion
RUN  apt-get install -y --no-install-recommends nodejs npm \
    && npm install -g @datadog/datadog-ci
```
Actually, with the switch to bookworm, we don't need to install from nodesource anyway.

**Review by avara1986 — APPROVED** [review]

**Comment by romainkomorndatadog** [comment]
This isn't needed anymore since the switch to bookworm.

**Pull request closed without merging**